torch/csrc/autograd/profiler.h

```diff
@@ -129,9 +132,33 @@ struct TORCH_API Event final {
         thread_id_(thread_id),
         shapes_(shapes) {
     record(record_cuda);
+    p_event_counter = new unsigned(1);
```

Comment:
This is a huge code smell to me. At the very least this should be a `unique_ptr` so you don't manually have to manage the memory, but the fact that we need to manually implement refcounting is extremely smelly and indicates some sort of larger architectural problem

Reply:
You mean `shared_ptr`? We can do something like 
```c++
p_event_counter = std::make_shared<unsigned>(1);
```
and avoid `delete p_event_counter;` in the destructor.